jobbergate-agent/tests/jobbergate/test_compute.py

```diff
@@ -0,0 +1,307 @@
+
+        measures = []
+        aggregated_data: dict[tuple[int, str, str, str], dict[str, float]] = {}
+
+        # Generate measures
+        for _ in range(num_points_per_measurement):
+            for host in range(1, num_hosts + 1):
+                for job in range(1, num_jobs + 1):
+                    for step in range(1, num_steps + 1):
+                        for task in range(1, num_tasks + 1):
+                            key = (current_time, f"host_{host}", str(step), str(task))
+
+                            if key not in aggregated_data:
+                                aggregated_data[key] = default_measurements.copy()
+
+                            for measurement in measurement_names:
+                                value = random.random() * 100
+                                measure = InfluxDBMeasure(
+                                    **{
+                                        "time": current_time,
+                                        "host": f"host_{host}",
+                                        "job": str(job),
+                                        "step": str(step),
+                                        "task": str(task),
+                                        "value": value,
+                                        "measurement": measurement,
+                                    }
+                                )
+                                measures.append(measure)
+
+                                # Aggregate value
+                                aggregated_data[key][measurement] = value
+                    current_time += 10
+
+        # Create aggregated list
+        aggregated_list = cast(
+            list[tuple[int, str, str, str, float, float, float, float, float, float, float, float, float, float]],
+            [
+                (
+                    time,
+                    host,
+                    step,
+                    task,
+                    *(aggregated_data[(time, host, step, task)][measurement] for measurement in measurement_names),
+                )
+                for (time, host, step, task) in aggregated_data
+            ],
+        )
+
+        return measures, aggregated_list
+
+    return _generate_and_aggregate
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "num_points_per_measurement, num_hosts, num_jobs, num_steps, num_tasks",
+    [
+        (1, 1, 1, 1, 1),
+        (3, 10, 1, 5, 10),
+        (7, 3, 1, 4, 2),
+    ],
+)
+async def test_aggregate_influx_measures__success(
+    num_points_per_measurement: int,
+    num_hosts: int,
+    num_jobs: int,
+    num_steps: int,
+    num_tasks: int,
+    generate_and_aggregate_job_metrics_data: Callable[
+        [int, int, int, int, int],
+        tuple[
+            list[InfluxDBMeasure],
+            list[tuple[int, str, str, str, float, float, float, float, float, float, float, float, float, float]],
+        ],
+    ],
```

Comment:
This type signature is too much. It's not helpful on its own; you still have to look at the implementation to understand it. Let's leave it off here.

Reply:
I created a TypeAlias for this since this is used multiple times in the code base. Let me know if it is enough :)

[cf0955b](https://github.com/omnivector-solutions/jobbergate/pull/663/commits/cf0955b169b1af76a0525107f37bc4cef3d478d6)

Reply:
That's useful, and the type hint isn't so onerous there. When we move away from using fixtures for these type of functions, this will naturally resolve itself.